core/embed/firmware/systemview.c

```diff
@@ -0,0 +1,115 @@
+#ifdef SYSTEM_VIEW
+
+#include "systemview.h"
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include "mpconfigport.h"
+
+#include "SEGGER_SYSVIEW.h"
+#include "SEGGER_SYSVIEW_Conf.h"
+
+#define SYSTICK ((SYSTICK_REGS *)0xE000E010)
+#define SCS ((SCS_REGS *)0xE000ED00)
+
+// for storing DWT CYCCNT from SVC call
+volatile uint32_t cyccnt_cycles;
+
+typedef struct {
+  volatile unsigned int CSR;
+  volatile unsigned int RVR;
+  volatile unsigned int CVR;
+  volatile unsigned int CALIB;
+} SYSTICK_REGS;
+
+typedef struct {
+  volatile unsigned int CPUID;  // CPUID Base Register
+  volatile unsigned int ICSR;   // Interrupt Control and State Register
+  volatile unsigned int VTOR;   // Vector Table Offset Register
+  volatile unsigned int
+      AIRCR;  // Application Interrupt and Reset Control Register
+  volatile unsigned int SCR;    // System Control Register
+  volatile unsigned int CCR;    // Configuration and Control Register
+  volatile unsigned int SHPR1;  // System Handler Priority Register 1
+  volatile unsigned int SHPR2;  // System Handler Priority Register 2
+  volatile unsigned int SHPR3;  // System Handler Priority Register 3
+  volatile unsigned int SHCSR;  // System Handler Control and State Register
+  volatile unsigned int CFSR;   // Configurable Fault Status Register
+  volatile unsigned int HFSR;   // HardFault Status Register
+  volatile unsigned int DFSR;   // Debug Fault Status Register
+  volatile unsigned int MMFAR;  // MemManage Fault Address Register
+  volatile unsigned int BFAR;   // BusFault Address Register
+  volatile unsigned int AFSR;   // Auxiliary Fault Status Register
+  volatile unsigned int aDummy0[4];  // 0x40-0x4C Reserved
+  volatile unsigned int aDummy1[4];  // 0x50-0x5C Reserved
+  volatile unsigned int aDummy2[4];  // 0x60-0x6C Reserved
+  volatile unsigned int aDummy3[4];  // 0x70-0x7C Reserved
+  volatile unsigned int aDummy4[2];  // 0x80-0x87 - - - Reserved.
+  volatile unsigned int CPACR;       // Coprocessor Access Control Register
+} SCS_REGS;
+
+extern uint32_t SystemCoreClock;
+
+static inline uint32_t is_mode_unprivileged(void) {
+  uint32_t r0;
+  __asm__ volatile("mrs %0, control" : "=r"(r0));
+  return r0 & 1;
+}
+
```

Comment:
Can be replaced with `#include "supervise.h"`

Reply:
supervise.h wasn't enough and there used to be linker conflict with duplicate symbols. Found some combination of includes that fixes it.